lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,281 @@
+    }
+  }
+
+  /**
+    * Show the canvas and hide original video element
+    */
+  showCanvas() {
+    if (this.dil_) {
+      this.media_.style.display = 'none';
+      this.canvas_.style.display = 'block';
+    }
+  }
+
+  /**
+   * Create LCEVC Dil.
+   * @private
+   */
+  createDil_() {
+    if (this.isSupported() && !this.dil_) {
+      if (!this.canvas_) {
+        this.createCanvas_();
+      }
+      if (LcevcDil.SupportObject.webGLSupport(this.canvas_)) {
+        this.dil_ = new LcevcDil.LcevcDil(
+            this.media_,
+            this.canvas_,
+            this.dilConfig_);
+
+        this.setControlsStyle_();
+      }
+    }
+  }
+
+  /**
+   * Reset LCEVC Dil.
+   */
+  async reset() {
+    const INVALID_LEVEL = -2;
+    if (this.dil_) {
+      this.dil_.setCurrentLevel(INVALID_LEVEL);
+      await Promise.resolve();
+      this.dil_.setLevelSwitching(this.level_, 1);
+    }
+  }
+
+  /**
+   * Close LCEVC Dil.
+   */
+  close() {
+    if (this.dil_) {
+      this.dil_.close();
+      this.dil_ = null;
+    }
+    this.lcevcContainer_.remove();
+    if (this.resizeObserver_) {
+      this.resizeObserver_.disconnect();
+      this.resizeObserver_ = null;
+    }
+    if (this.canvas_) {
+      this.canvas_.remove();
+    }
+  }
+
+  /**
+   * Check if LCEVC Dil exists.
+   */
+  isSupported() {
+    if (typeof libDPIModule === 'undefined') {
+      shaka.log.alwaysWarn(
+          'Could not Find LCEVC Library dependencies on this page');
+    }
+
+    if (typeof LcevcDil === 'undefined') {
+      shaka.log.alwaysWarn('Could not Find LCEVC Library on this page');
+    } else {
+      if (!LcevcDil.SupportStatus) {
+        shaka.log.alwaysWarn(LcevcDil.SupportError);
+      }
+    }
+
+    return typeof LcevcDil !== 'undefined' &&
+      typeof libDPIModule !== 'undefined' &&
+      LcevcDil.SupportStatus;
+  }
+
+  /**
+   * Create canvas for LCEVC Dil to render if not exists.
+   * @private
+   */
+  createCanvas_() {
+    this.canvas_ = /** @type {HTMLCanvasElement} */
+      (shaka.util.Dom.createHTMLElement('canvas'));
+    if (LcevcDil.SupportObject.webGLSupport(this.canvas_)) {
+      this.lcevcContainer_ = shaka.util.Dom.createHTMLElement('div');
+      this.lcevcContainer_.style.position = 'relative';
+
+      if (this.dilConfig_['playerControls']) {
+        this.videoControls_ = shaka.util.Dom.getElementById(
+            this.dilConfig_['playerControls']['controlsID']);
+        this.videoControls_.id =
+        this.dilConfig_['playerControls']['controlsID'];
+        this.lcevcContainer_.appendChild(this.videoControls_);
+      }
+
+      this.canvas_.style.position = 'absolute';
+      this.media_.style.display = 'none';
+
+      this.parent_ = this.media_.parentElement;
+      if (this.parent_) {
+        this.lcevcContainer_.appendChild(this.canvas_);
+        this.parent_.appendChild(this.lcevcContainer_);
+        this.resizeObserver_ = new ResizeObserver((_entries) => {
```

Comment:
ResizeObserver isn't available on all devices, so you should add a check for this in `isSupported()`.

Reply:
Check for resize observer is added based on availability  and the function is called separately once on setup.